### PR TITLE
parallel execution for getting status of services in an environment

### PR DESCRIPTION
**Pull request by boppanasusanth**
# Odin version(x.y.z)

### CODE REVIEW

File: internal/backend/env.go [hunk 1]
```diff
@@ -141,11 +142,18 @@ func (e *Env) EnvServiceStatus(env, serviceName string) (envResp.EnvServiceStatu
 	client := newApiClient()
 
 	response := client.actionWithRetry(path.Join(envEntity, env)+"/services/"+serviceName+"/status", "GET", nil)
-	response.Process(true) // process response and exit if error
+	// response.Process(true) // process response and exit if error
```
Comment (surajgour-d11) [inline internal/backend/env.go:145]: remove comment

File: internal/backend/env.go [hunk 2]
```diff
@@ -141,11 +142,18 @@ func (e *Env) EnvServiceStatus(env, serviceName string) (envResp.EnvServiceStatu
...
 
 	var envResponse envResp.EnvServiceStatusResponse
 	err := json.Unmarshal(response.Body, &envResponse)
 
+	if response.Error != nil {
+		return envResponse.ServiceResponse, response.Error
+	}
+	if response.StatusCode >= 400 && response.StatusCode < 600 {
```
Comment (surajgour-d11) [inline internal/backend/env.go:153]: instead check if response is not 2XX

File: internal/backend/env.go (hunk 1 above)
Comment (boppanasusanth) [reply internal/backend/env.go]: done

File: internal/backend/env.go (hunk 2 above)
Comment (boppanasusanth) [reply internal/backend/env.go]: done

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -81,3 +82,19 @@ func FetchKey(keyName string) string {
 	f := reflect.Indirect(r).FieldByName(keyName)
 	return f.String()
 }
+
+func MaxWidth(a, b int) int {
```
Comment (surajgour-d11) [inline pkg/utils/utils.go:86]: you can use `math.Max`

File: pkg/table/table.go [hunk 4]
```diff
@@ -37,6 +37,49 @@ func Write(headers []string, data [][]interface{}) {
 	table.Render()
 }
 
+func AppendRow(row []interface{}, colWidth []int) {
+	table := tablewriter.NewWriter(os.Stdout)
+
+	table.SetRowLine(false)
```
Comment (surajgour-d11) [inline pkg/table/table.go:43]: wrap repeating table configuration in a function

File: pkg/utils/utils.go (hunk 3 above)
Comment (boppanasusanth) [reply pkg/utils/utils.go:86]: `math.Max` takes float64 as input. there is no inbuilt max function that takes integers as input

File: pkg/table/table.go (hunk 4 above)
Comment (boppanasusanth) [reply pkg/table/table.go:43]: okay.

Comment (boppanasusanth) [reply pkg/table/table.go]: done. please check

**Review by surajgour-d11 — APPROVED** [review]

File: internal/backend/env.go [hunk 5]
```diff
@@ -141,11 +142,17 @@ func (e *Env) EnvServiceStatus(env, serviceName string) (envResp.EnvServiceStatu
...
-	response.Process(true) // process response and exit if error
 
 	var envResponse envResp.EnvServiceStatusResponse
 	err := json.Unmarshal(response.Body, &envResponse)
 
+	if response.Error != nil {
+		return envResponse.ServiceResponse, response.Error
+	}
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		return envResponse.ServiceResponse, errors.New(string(response.Body))
```
Comment (akshaypatidar1999) [inline internal/backend/env.go:153]: Use `Process` method from `pkg/request/request.go`

**Review by akshaypatidar1999 — APPROVED** [review]

**Pull request merged**